cmd/erasure-multipart.go

```diff
@@ -716,11 +704,26 @@ func (er erasureObjects) PutObjectPart(ctx context.Context, bucket, object, uplo
 		index = opts.IndexCB()
 	}
 
+	var actualSize int64
+	_, encrypted := crypto.IsEncrypted(fi.Metadata)
+	compressed := fi.IsCompressed()
+	switch {
+	case compressed:
+		actualSize = data.ActualSize()
+	case encrypted:
+		decSize, err := sio.DecryptedSize(uint64(n))
+		if err == nil {
+			actualSize = int64(decSize)
+		}
+	default:
```

Comment:
shouldn't we add `case compressed && encrypted:` ? 

it should be possible to calculate it,  I think we compress then we encrypt

Reply:
you can't @vadmeste this is borrowed from the idea in cmd/object-api-utils.go